src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ConfigBuilderCacheWarmer.php

```diff
@@ -47,14 +47,17 @@ public function __construct(KernelInterface $kernel, ?LoggerInterface $logger =
      */
     public function warmUp(string $cacheDir)
     {
-        $generator = new ConfigBuilderGenerator($this->kernel->getBuildDir());
+        /** @var ContainerBuilder $container */
+        $container = \Closure::bind(function (KernelInterface $kernel) {
```

Comment:
psalm is right, we're hardcoding the logic against Kernel:
```suggestion
        $container = \Closure::bind(function (Kernel $kernel) {
```

Reply:
would it make sense to accept kernels that don't extend Kernel BTW? (using the previous logic for them)

Reply:
Done, though Psalm still isn't happy. I've also added a case for kernels not extending the Kernel class.